indico/MaKaC/common/output.py

```diff
@@ -1365,26 +1351,26 @@ def resourceToXMLMarc21(self, res, out=None):
             self.resourceLinkToXMLMarc21(res, out=out)
 
     def _attachment_unique_id(self, attachment, add_prefix=True):
-        unique_id = uniqueId(attachment.folder.linked_object)
+        unique_id = str(uniqueId(attachment.folder.object))
         if add_prefix:
             unique_id = "INDICO." + unique_id
         if attachment.legacy_mapping:
             unique_id += "m{}.{}".format(attachment.legacy_mapping.material_id, attachment.legacy_mapping.resource_id)
         else:
-            unique_id += "a{}".format(attachment.id)
+            unique_id += "a{}".format(str(attachment.id))
 
         return unique_id
 
     def _attachment_access_list(self, attachment):
-        linked_object = attachment.folder.linked_object
-        manager_list = set(linked_object.getRecursiveManagerList())
+        linked_object = attachment.folder.object
+        manager_list = set(linked_object.as_legacy.getRecursiveManagerList())
```

Comment:
you need to reimplement this (in `ProtectionManagersMixin`); managers are not stored in legacy code anymore